// gulp-htmlmin-jinjia2/index.js
'use strict';

var through = require('through2');
var gutil = require('gulp-util');
var minify = require('html-minifier').minify;
var extend= require('extend');

module.exports = function (opts) {

    opts = extend({}, {
        removeComments: true,                   //清除HTML注释
        collapseWhitespace: true,               //压缩HTML
        collapseBooleanAttributes: true,        //省略布尔属性的值 <input checked="true"/> ==> <input />
        removeEmptyAttributes: true,            //删除所有空格作属性值 <input id="" /> ==> <input />
        removeScriptTypeAttributes: true,       //删除<script>的type="text/javascript"
        removeStyleLinkTypeAttributes: true,    //删除<style>和<link>的type="text/css"
        minifyJS: true,                         //压缩页面JS
        minifyCSS: true,                        //压缩页面CSS
        ignoreCustomFragments: [                //不处理 {} {%%} 里的内容
            /\{\%[\s\S]*?\%\}/g,
            /\{\{[\s\S]*?\}\}/g
        ]
    }, opts);


    function main(file, encoding, callback){

        if (file.isNull()) {
            cb(null, file);
        }

        var html = file.contents.toString();
        html = minify(html, opts);
        file.contents = new Buffer(html);

        this.push(file);
        callback();
    }

    return through.obj(main);
};
